src/arlo_plugin/mvss.py

```diff
@@ -0,0 +1,217 @@
+"""Created new mvss.py based on existing vss.py and cleaned out code that was not needed for the siren."""
+from __future__ import annotations
+
+import asyncio
+from typing import List, TYPE_CHECKING
+
+from scrypted_sdk.types import Device, DeviceProvider, Setting, Settings, SettingValue, SecuritySystem, SecuritySystemMode, Readme, ScryptedInterface, ScryptedDeviceType
+
+from .base import ArloDeviceBase
+from .util import async_print_exception_guard
+
+if TYPE_CHECKING:
+    # https://adamj.eu/tech/2021/05/13/python-type-hints-how-to-fix-circular-imports/
+    from .provider import ArloProvider
+
+class ArloModeVirtualSecuritySystem(ArloDeviceBase, SecuritySystem, Settings, Readme, DeviceProvider):
+    """A virtual, emulated security system that controls the Routines Mode in the Arlo App."""
+
+    SUPPORTED_MODES = [SecuritySystemMode.AwayArmed.value, SecuritySystemMode.HomeArmed.value, SecuritySystemMode.Disarmed.value]
+
+    def __init__(self, nativeId: str, arlo_device: dict, arlo_basestation: dict, provider: ArloProvider, parent: str) -> None:
+        super().__init__(nativeId=nativeId, arlo_device=arlo_device, arlo_basestation=arlo_basestation, provider=provider)
+        """Sets the parent as the nativeId passed through the call from provider.py."""
+        self.parent = parent
+        self.create_task(self.delayed_init())
+
+    """Storage Location for holding the location code required to pass back on the put request."""
+    @property
+    def mode_location(self) -> str:
+        mode_location_revision = self.provider.arlo.GetMode()
+        mode_location = mode_location_revision[1]
+        return mode_location
+
+    @mode_location.setter
+    def mode_location(self, mode_location: str) -> None:
+        self.storage.setItem("mode_location", mode_location)
+
+
+    """Storage Location for holding the revision code required to pass back on the put request."""
+    @property
+    def mode_revision(self) -> str:
```

Comment:
perhaps name this `next_mode_revision`? since `mode_revision` implies the current revision number